common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -0,0 +1,117 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+
+/**
+ * This class wraps a JAAS Subject and provides methods to determine the user's name. It supports
+ * Windows, Unix, and Kerberos login modules.
+ */
+public class UserInformation {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static final String OS_LOGIN_MODULE_NAME;
+  private static final Class<? extends Principal> OS_PRINCIPAL_CLASS;
+  private static final boolean WINDOWS = System.getProperty("os.name").startsWith("Windows");
+  private static final boolean IS_64_BIT = System.getProperty("os.arch").contains("64");
+  private static final boolean AIX = System.getProperty("os.name").equals("AIX");
+  public static final String JAVA_VENDOR_NAME = System.getProperty("java.vendor");
+  public static final boolean IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM");
+
+  static {
+    OS_LOGIN_MODULE_NAME = getOSLoginModuleName();
+    OS_PRINCIPAL_CLASS = getOsPrincipalClass();
+  }
+
+  // Return the OS login module class name.
+  private static String getOSLoginModuleName() {
+    if (IBM_JAVA) {
+      if (WINDOWS) {
+        return IS_64_BIT ? "com.ibm.security.auth.module.Win64LoginModule"
+            : "com.ibm.security.auth.module.NTLoginModule";
+      } else if (AIX) {
+        return IS_64_BIT ? "com.ibm.security.auth.module.AIX64LoginModule"
+            : "com.ibm.security.auth.module.AIXLoginModule";
+      } else {
+        return "com.ibm.security.auth.module.LinuxLoginModule";
+      }
+    } else {
+      return WINDOWS ? "com.sun.security.auth.module.NTLoginModule"
+          : "com.sun.security.auth.module.UnixLoginModule";
+    }
+  }
+
+  // Return the OS principal class
+  static Class<? extends Principal> getOsPrincipalClass() {
+    ClassLoader cl = ClassLoader.getSystemClassLoader();
+    try {
+      String principalClass = null;
+      if (IBM_JAVA) {
+        if (IS_64_BIT) {
+          principalClass = "com.ibm.security.auth.UsernamePrincipal";
+        } else {
+          if (WINDOWS) {
+            principalClass = "com.ibm.security.auth.NTUserPrincipal";
+          } else if (AIX) {
+            principalClass = "com.ibm.security.auth.AIXPrincipal";
+          } else {
+            principalClass = "com.ibm.security.auth.LinuxPrincipal";
+          }
+        }
+      } else {
+        principalClass = WINDOWS ? "com.sun.security.auth.NTUserPrincipal"
+            : "com.sun.security.auth.UnixPrincipal";
+      }
+      return (Class<? extends Principal>) cl.loadClass(principalClass);
+    } catch (ClassNotFoundException e) {
```

Comment:
Nice catch! 
It maybe better to just return a class name, and when using it later, the caller handles this load exception. (for example, the upcoming TACHYON_LOGIN_MODULE).